OracleTuxedo/core/samples/ws_ssl_svr/ws_svr_runme.sh

```diff
@@ -0,0 +1,167 @@
+echo "CA_PASSWORD = ${CA_PASSWORD} during creation of the CA key and certificates."
+openssl req -new -x509 -extensions v3_ca -keyout private/ca_key.pem -out certs/ca_cert.pem \
+            -days 3650 -config ./openssl.cnf  \
+            -passout pass:"${CA_PASSWORD}"  \
+            -subj "/C=US/ST=CA/L=SV/O=OraTux/OU=IT/CN=ISH_tuxsvr"
+
+# create tux server key and tux server certificate signing request (CSR)
+openssl req -new -nodes -out tuxsvr-req.pem -keyout private/tuxsvr_key.pem \
+            -config ./openssl.cnf \
+            -subj "/C=US/ST=CA/L=SV/O=OraTux/OU=IT/CN=ISH_tuxsvr"
+
+# sign the tux server CSR using CA cert to create the tux server certificate
+openssl ca -out certs/tuxsvr_cert.pem -config ./openssl.cnf -passin pass:"${CA_PASSWORD}" \
+           -batch -infiles tuxsvr-req.pem
+
+# create the PKCS12 wallet
+cat certs/tuxsvr_cert.pem certs/ca_cert.pem private/tuxsvr_key.pem > pre-p12.pem
+openssl pkcs12 -export -in pre-p12.pem -password pass:"${PASSVAR}"  -out "${WALLET_DIR}"/ewallet.p12
+echo "Created ewallet.p12 in ${WALLET_DIR}"
+
+cd "${APPDIR}" || return 1
+# "create ssl certificates and PKCS12 wallet" - end
+
+
+# clean up from any previous run
+tmshutdown -y &>/dev/null 
+rm -Rf serverws simpserv tuxconfig ubbws ULOG.*
+
+# Create the Tuxedo configuration file
+cat >ubbws << EOF
+*RESOURCES
+IPCKEY          $IPCKEY
+DOMAINID        ws_svr
+MASTER          site1
+MAXACCESSERS    50
+MAXSERVERS      20
+MAXSERVICES     10
+MODEL           SHM
+LDBAL           Y
+SECURITY        NONE
+
+*MACHINES
+"$HOSTNAME" LMID=site1
+            APPDIR="$APPDIR"
+            TUXCONFIG="$APPDIR/tuxconfig"
+            TUXDIR="$TUXDIR"
+            MAXWSCLIENTS=5
+            SEC_PRINCIPAL_NAME="$SEC_PRINCIPAL_NAME"
+            SEC_PRINCIPAL_LOCATION="$APPDIR/wallet"
+            SEC_PRINCIPAL_PASSVAR="PASSVAR"
+
+*GROUPS
+APPGRP      LMID=site1  GRPNO=1  OPENINFO=NONE
+WSGRP       LMID=site1  GRPNO=2  OPENINFO=NONE
+
+*SERVERS
+serverws    SRVGRP=APPGRP  SRVID=1  CLOPT="-A"
+simpserv    SRVGRP=APPGRP  SRVID=3  CLOPT="-A"
+WSL         SRVGRP=WSGRP  
+            SRVID=10  
+            CLOPT="-A -- -n ${WSNADDR} -m 2 ${NATADDR_OPT} -p 2071 -P 2075 -S ${SSL_PORT} -R 3600"
+            RESTART=Y     
+            MAXGEN=10 
+            GRACE=3600
+
+*SERVICES
+BASICWS
+TOUPPER
+
+EOF
+
+# Get the sources if not already in this directory
+if [ ! -r serverws.c ]
+    then
+    cp /u01/oracle/serverws.c .
+fi
+if [ ! -r simpserv.c ]
+    then
+    cp /u01/oracle/simpserv.c .
+fi
+
+# Compile the configuration file and build the client and server
+# "cat /dev/null" below makes "tmloadcf" take the password from "PASSVAR"
+# shellcheck disable=SC2002
+cat /dev/null | tmloadcf -y ubbws
+buildserver -o serverws -f serverws.c -s BASICWS
+buildserver -o simpserv -f simpserv.c -s TOUPPER
+
+# Boot up the domain
+tmboot -y
+
+cat "${APPDIR}"/ULOG*
+
+while true
+do
+    ONE_HOUR=3600
+    sleep ${ONE_HOUR}
+done
```

Comment:
Why are you only allowing the container to run for an hour? And why are you setting an environment variable to do this? Can you explain what the intention behind this is?

Reply:
Its not for one hour. Its an infinite sleep but wakes up every one hour. Without this loop the Tuxedo server will shut down because in Kubernetes the container exits immediately. This app has been created mostly to run a sample in Kubernetes cluster.

Reply:
Why can't you run the Tuxedo server in the foreground so it doesn't close the container?

Reply:
We had to do something similar for WebLogic as running in the foreground isn't practical as there are so many customer-visible behaviors in the existing scripts. That said, sleeping for an hour will cause problems when you are in a container orchestration environment (like Kubernetes) because your container won't shut down promptly but will instead always take the maximum time configured for the deletion. 

We used a shorter sleep combined with a marker file that was created when the server exited: https://github.com/oracle/weblogic-kubernetes-operator/blob/main/operator/src/main/resources/scripts/utils.sh#L567

Reply:
Thanks to both of you. I added a similar SHUTDOWN_MARKER_FILE to our startup script too now and with the sleep wakeup interval reduced to 1 second now.